website/source/docs/configuration/providers.html.md

```diff
@@ -117,3 +117,14 @@ KEY {
 	CONFIG
 }
 ```
+
+## Interpolation
+Providers support [interpolation syntax](/docs/configuration/interpolation.html) allowing dynamic configuration at run time.
+
+```
+provider "aws" {
+	region = "${var.aws_region}"
+}
+```
+
+Only [variables](/docs/configuration/variables) and [remote state](/docs/state/remote.html) are supported, it is not possible to use the output from a resource, module or data source in the interpolation syntax for a provider.
```

Comment:
I agree with @radeksimko - I think we can change the tone a bit here, and do some education in the process. Consider:

Because providers are one of the first things loaded when Terraform parses the graph, it is not possible to use the output from modules, resources, or data sources as inputs to the provider. At this time, only [variables](link) and [remote state](link) may be used in an interpolation inside a `provider` stanza.